rcl_action/src/rcl_action/action_server.c

```diff
@@ -189,12 +203,16 @@ rcl_action_server_init(
   }
 
   // Copy action name
-  action_server->impl->action_name = rcutils_strdup(action_name, allocator);
+  action_server->impl->action_name = rcutils_strdup(resolved_action_name, allocator);
```

Comment:
Change this will action_server_get_action_name().  This is why you also update test codes.   
I refer to the implementation of other functions rcl_client_get_service_name/rcl_service_get_service_name. They all return original service name (Not remapped).   
So I prefer not to modify action_name here. Is it necessary to add a new item to store the resolved action name?

Reply:
I looked into the implementation of `rcl_client_get_service_name()`/`rcl_service_get_service_name()`, they both return `impl->rmw_handle->service_name` which is the remapped and expanded name. I have tested this behavior by modifying the service and client rclc examples to accept remaps from the cmd line. Using `rcl_{client,service}_get_service_name()` returned the fully expanded and remapped service name in both cases.

https://github.com/ros2/rcl/blob/0723613c89d86f23a4aef70601994b812a6d15a2/rcl/src/rcl/service.c#L295-L304

`rcl_action_server_get_action_name()` returns `action_server->impl->action_name`, I think it would be more consistent with the services if this is also the remapped and expanded name. Maybe we could rename the field to `remapped_action_name` to make it clearer?

Reply:
You are right. It‘s my fault.  

> Maybe we could rename the field to remapped_action_name to make it clearer?

Totally agree.

Reply:
You are right. It's my fault.

> Maybe we could rename the field to remapped_action_name to make it clearer?

Totally agree.